tests/integration/pilot/tunneling/tunneling_test.go

```diff
@@ -0,0 +1,416 @@
+		ctx.ConfigIstio().EvalFile(meshNs.Name(), templateParams, "forward-proxy/destination-rule.tmpl.yaml").ApplyOrFail(ctx)
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "forward-proxy/ssl-certificate-configmap.yaml")
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "forward-proxy/ssl-private-key-configmap.yaml")
+		ctx.ConfigIstio().EvalFile(externalNs.Name(), templateParams, "forward-proxy/configmap.tmpl.yaml").ApplyOrFail(ctx)
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "forward-proxy/deployment.yaml")
+		waitForPodsReadyOrFail(ctx, externalNs.Name(), "external-forward-proxy")
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		common.ApplyFileOrFail(ctx, meshNs.Name(), "external-app/service-entry.yaml")
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "external-app/ssl-certificate-configmap.yaml")
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "external-app/ssl-private-key-configmap.yaml")
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "external-app/configmap.yaml")
+		common.ApplyFileOrFail(ctx, externalNs.Name(), "external-app/deployment.yaml")
+		waitForPodsReadyOrFail(ctx, externalNs.Name(), "external-app")
+	}()
+
+	wg.Wait()
+	makeExternalServicesResolvable(ctx, externalNs.Name(), meshNs.Name())
+
+	for _, tc := range testCases() {
+		for _, res := range tc.istioResourcesToApply {
+			common.ApplyFileOrFail(ctx, meshNs.Name(), res)
+		}
+
+		for _, protocol := range tc.protocols {
+			ctx.NewSubTest(fmt.Sprintf("%s/%s-request", tc.name, protocol)).Run(func(ctx framework.TestContext) {
+				// requests will fail until istio-proxy gets the Envoy configuration from istiod, so retries are necessary
+				retry.UntilSuccessOrFail(ctx, func() error {
+					if err := executeRequestToExternalApp(ctx, meshNs.Name(), protocol, tc.name); err != nil {
+						return err
+					}
+					externalForwardProxyIP := getPodIP(ctx, externalNs.Name(), "external-forward-proxy")
+					if err := verifyThatRequestWasTunneled(ctx, externalNs.Name(), externalForwardProxyIP, protocol, tc.name); err != nil {
+						return err
+					}
+					return nil
+				}, retry.Timeout(10*time.Second), retry.Delay(1*time.Second))
+			})
+		}
+
+		for _, res := range tc.istioResourcesToApply {
+			common.DeleteFileOrFail(ctx, meshNs.Name(), res)
+		}
+		// make sure that configuration changes were pushed to sidecar proxies
+		time.Sleep(10 * time.Second)
+	}
+}
+
+func executeRequestToExternalApp(ctx framework.TestContext, meshNs, protocol, testName string) error {
+	kubeClient := ctx.Clusters().Default()
+	sleepPodName := getPodName(ctx, meshNs, "sleep")
+
+	testRequestCmd := fmt.Sprintf(
+		// The command below performs a GET request and writes only HTTP status code to the output.
+		// Flag --insecure must be passed, because self-signed certificates are used.
+		"curl --insecure -s -o /dev/null -w '%%{http_code}' %s://external-app.testdomain:%d/test/%s",
```

Comment:
Should use the echo app instead of exec+curl

Reply:
Ok, I will try to replace it with echo.

Reply:
Done